New features list (#33)
meilisearch/documentation: introduction/features.md

## Features

Our feature list with extended explanations and links to the documentation.

> *  **Instant Search** (answers < 50ms): Priority on fast answers for smooth search experience.

Reviewer: ```suggestion
* **Instant Search** (answers < 50ms): Priority on fast answers for smooth search experience.
```

> * **Search as you type** (*prefix search*): Results are shown as you type. To achieve this, we use a [prefix-search](/advanced_guides/prefix.html#prefix-search).

Reviewer 2: ```suggestion
* **Search as you type** (*prefix search*): Results are updated on each keystroke. We use a [prefix-search](/advanced_guides/prefix.html#prefix-search) for the best results.
```

🤷‍♀

Author: I wanted to highlight that the instant search was made possible by our use of a prefix-search. And not that the prefix-search made for best results.

* [Typo tolerance](/advanced_guides/typotolerance.md#typo-tolerance): Understands typo and spelling mistakes.
* [Tokenization](https://en.wikipedia.org/wiki/Lexical_analysis#Tokenization) in English, kanji and latin based languages.  
* Ability to create [synonyms](/advanced_guides/synonyms.md) for a better search experience.   
* [Easy to install, deploy and maintain](/introduction/quickstart.html#quick-start)
* **Return the whole document**: The entire document is returned upon search.
* **Highly customizable search and indexation**:
    - [Filters](/advanced_guides/search_parameters.md#filters): Filter in search
    - [Highlight](/advanced_guides/search_parameters.md#attributes-to-highlight): Highlighted search results in documents

>     - [Maximum response time](/advanced_guides/search_parameters.md#attributes-to-highlight): Never will a response take more time to arrive

Reviewer 2: I am not this feature exists today

Author: Well, it exists, but it doesn't work. I'll come back to this request once we know when it will be fixed
